### ISSUE TITLE
select from a Table-Valued Function (TVF) 

### ISSUE THREAD

**Issue by enghussein** [dotnet/efcore#8013]
i want to make select query from sql server table-value-function from and i am using EF7, ASP.NET Core RC1

SELECT * FROM [dbo].[GetMyFunction] ( 2017,1)
i created fake entity as the following :
```
public class MyFunctionResult
{

    public short CountryCode { get; set; }
    public string CountryName { get; set; }
}

 public IQueryable<MyFunctionResult> CallMyFunction()
    {
        string query = string.Format("select * from [dbo].[GetMyFunction] ({0},{1})", 2017, 1);
          return     _dbContext.Set<MyFunctionResult>().FromSql(query);
   }


 public virtual DbSet<MyFunctionResult> MyFunctionResults { get; set; }
 protected override void OnModelCreating(ModelBuilder modelBuilder)
{
modelBuilder.Ignore<MyFunctionResult>();
}
```
Exception message:   "Value cannot be null.\r\nParameter name: entityType"
Stack trace:
"   at Microsoft.Data.Entity.Utilities.Check.NotNull[T](T value, String parameterName)\r\n   at Microsoft.Data.Entity.SqlServerMetadataExtensions.SqlServer(IEntityType entityType)\r\n   at Microsoft.Data.Entity.Metadata.SqlServerAnnotationProvider.For(IEntityType entityType)\r\n   at Microsoft.Data.Entity.Query.ExpressionVisitors.RelationalEntityQueryableExpressionVisitor.VisitEntityQueryable(Type elementType)\r\n   at Microsoft.Data.Entity.Query.ExpressionVisitors.EntityQueryableExpressionVisitor.VisitConstant(ConstantExpression constantExpression)\r\n   at System.Linq.Expressions.ConstantExpression.Accept(ExpressionVisitor visitor)\r\n   at Microsoft.Data.Entity.Query.ExpressionVisitors.ExpressionVisitorBase.Visit(Expression expression)\r\n   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.ReplaceClauseReferences(Expression expression, IQuerySource querySource, Boolean inProjection)\r\n   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.CompileMainFromClauseExpression(MainFromClause mainFromClause, QueryModel queryModel)\r\n   at Microsoft.Data.Entity.Query.RelationalQueryModelVisitor.CompileMainFromClauseExpression(MainFromClause mainFromClause, QueryModel queryModel)\r\n   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.VisitMainFromClause(MainFromClause fromClause, QueryModel queryModel)\r\n   at Remotion.Linq.Clauses.MainFromClause.Accept(IQueryModelVisitor visitor, QueryModel queryModel)\r\n   at Remotion.Linq.QueryModelVisitorBase.VisitQueryModel(QueryModel queryModel)\r\n   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.VisitQueryModel(QueryModel queryModel)\r\n   at Microsoft.Data.Entity.Query.RelationalQueryModelVisitor.VisitQueryModel(QueryModel queryModel)\r\n   at Microsoft.Data.Entity.Query.Internal.SqlServerQueryModelVisitor.VisitQueryModel(QueryModel queryModel)\r\n   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.CreateQueryExecutor[TResult](QueryModel queryModel)\r\n   at Microsoft.Data.Entity.Storage.Database.CompileQuery[TResult](QueryModel queryModel)\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at Microsoft.Data.Entity.Query.Internal.QueryCompiler.<>c__DisplayClass18_0`1.<CompileQuery>b__0()\r\n   at Microsoft.Data.Entity.Query.Internal.CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler)\r\n   at Microsoft.Data.Entity.Query.Internal.QueryCompiler.CompileQuery[TResult](Expression query)\r\n   at Microsoft.Data.Entity.Query.Internal.QueryCompiler.Execute[TResult](Expression query)\r\n   at Microsoft.Data.Entity.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression)\r\n   at Remotion.Linq.QueryableBase`1.GetEnumerator()\r\n   at System.Linq.SystemCore_EnumerableDebugView`1.get_Items()"
```



### Further technical details
EF Core version:7.0.0-rc1-final"
Database Provider: . EntityFramework.MicrosoftSqlServer
Operating system: windows
IDE: (e.g. Visual Studio 2015)

**Comment by smitpatel** [comment]
Your fake entity `MyFunctionResult` is not part of the model. At present EF does not allow to use FromSql returning non-entity objects. Tracking issue https://github.com/aspnet/EntityFramework/issues/1862

If you add `MyFunctionResult` in your model then the query would work.

Also, you are using really old pre-release version. We changed our name from EF 7 to EF Core long time ago and we have released RTM package last year. Please upgrade your EF version to latest RTM version for better EF.

**Comment by enghussein** [comment]
thx smitpatel for your replay , 
but what if i wont add it in the db? i mean i added it to the model ,but when i  want to skip this enity in migration "modelBuilder.Ignore<MyFunctionResult>(); " it will not work!
if i remove this skip "modelBuilder.Ignore<MyFunctionResult>();" it will work

about old version , its our next plan  to upgrade my project to RTM  version ,  but its not that easy to do it now since we are working in a big project

**Comment by smitpatel** [comment]
You can remove `CreateTable` call from scaffolded migration for that particular entity. That is the only way to get around creating migration at the moment. After #1862 is implemented, you will not need to add it as an entity type in the model.

**Comment by ajcvickers** [comment]
Dupe of #1862
